arangod/Cluster/ClusterInfo.cpp

```diff
@@ -949,6 +1050,26 @@ ClusterInfo::CollectionWithHash ClusterInfo::buildCollection(
         // we do have a view. set hash to 0, which will disable the caching
         // optimization
         hash = 0;
+        if (cleanupLinks) {
+          for (auto const& idx : indexes) {
+            TRI_ASSERT(idx);
+            if (idx->type() == Index::TRI_IDX_TYPE_IRESEARCH_LINK) {
+              auto& coordLink =
```

Comment:
Are we sure here that this code is only executed on coordinators?
Can we add an assert in this function to ensure this?